### PR TITLE
synch action getTables and metadata manifest

### PR DESCRIPTION
**Pull request by pivnicek**
This PR is mainly to provide the frontend with useful information about the DB in order to improve UX by allowing users to select tables/columns without having to write a query.
It also writes a metadata manifest so that the destination tables are appropriately decorated with type information etc...

Update components:
- [x] pgsql
- [x] mysql
- [x] mssql
- [x] db2
- [x] oracle
- [x] redshift
- [x] snowflake

### Round 2 

- [x] add schema names
- [x] add table-level metadata to manifest
- [x] add manifest tests to above repos 

### Final check

- [x] pgsql    
- [x] mysql    
- [x] mssql
- [ ] db2
- [ ] oracle
- [x] redshift
- [x] snowflake

### CODE REVIEW

File: src/Keboola/DbExtractor/Extractor/Common.php [hunk 1]
```diff
@@ -32,12 +32,21 @@ public function createConnection($params)
...
     public function testConnection()
     {
         $this->db->query("SELECT 1");
     }
+
+    public function showTables()
+    {
+        $tables = [];
+        $res = $this->db->query("SHOW TABLES");
+        while ($table = $res->fetch(\PDO::FETCH_NUM)) {
```
Comment (Halama) [inline src/Keboola/DbExtractor/Extractor/Common.php]: woudln't be more safe for feature extensions return array of objects?
something like:
```
[
 {
  "name": "escaping"
 },
 {
  "name": "another"
 }
]
```

Comment (pivnicek) [inline src/Keboola/DbExtractor/Extractor/Common.php]: yes, I think that's a valid point.  I'll be adding `describeTable` also so these table objects can fill out with that.  Thanks!

**Comment by Halama** [comment]
@pivnicek can you please describe what is the purpose of this PR and link it to some issue and PB feature.

**Comment by pivnicek** [comment]
Sure!

Issue 1) https://github.com/keboola/db-extractor-common/issues/9
PB 1)  https://keboola.productboard.com/feature-board/6271-planning/features/368835

PB 2) https://keboola.productboard.com/feature-board/6271-planning/features/268346

**Comment by pivnicek** [comment]
I'll update the description ^^ now

**Comment by Halama** [comment]
great, thanks.

**Comment by pivnicek** [comment]
Not at all, should have been done from the start, thanks for the push:)

**Comment by Halama** [comment]
ok, so now the first part sync action is done in this PR. and the second write a metadata manifest  will also be part of this OR?

**Comment by pivnicek** [comment]
They are now together, but I can remove the manifest portion and supply it separately if preferable.

**Comment by Halama** [comment]
no it is ok, It wasn't just clear for me if there are  only sync actions.

**Pull request merged**